doc/rtd/topics/datasources/ec2.rst

```diff
@@ -102,4 +110,10 @@ Notes
    The check for the instance type is performed by is_classic_instance()
    method.
 
+ * For EC2 instances with multiple network interfaces (NICs) attached, dhcp4
+   will be enabled to obtain the primary private IPv4 address of those NICs.
+   Wherever dhcp4 or dhcp6 is enabled for a NIC, a dhcp route-metric will be
+   added with the value of ``<interface_num + 1> * 100`` to ensure dhcp
```

Comment:
s/interface_num/device-number/

This corresponds with the ec2 metadata key 'device-number'.  We should also document that we consider 'primary interface' to be the nic config which has 'device-number: 0'.